### PR TITLE
bed file with states and/or variable binsizes to multivec

### PR DESCRIPTION
**Pull request by burakalver**
Here are the changes, also listed in the changelog.
- new option to import a "states" file format, a bed file with categorical data, e.g. from chromHMM to multivec format.
- while converting a bed file to multivec, each segment can now be a multiple of base_resolution,
rather than exactly match the base_resolution

And you can test it by:
`clodius convert bedfile_to_multivec test/sample_data/states_format_input_testfile.bed.gz --format states --row-infos-filename test/sample_data/states_format_test_row_infos.txt --assembly mm9 --starting-resolution 200 --num-rows 10`

### CODE REVIEW

File: clodius/cli/convert.py [hunk 1]
```diff
@@ -39,9 +39,40 @@ def epilogos_bedline_to_vector(bedline):
...
     return (chrom, start, end, states)
 
+def states_bedline_to_vector(bedline,states_dic):
+    '''
+    Convert a line from a bedfile containing states in categorical data to vector format.
+
+    Parameters
+    ----------
+
+    parts: [string,...]
```
Comment (pkerpedjiev) [inline clodius/cli/convert.py]:
```suggestion
    bedline: [string,...]
```
`bedline` is the name of the parameter, right?

File: clodius/cli/convert.py [hunk 2]
```diff
@@ -39,9 +39,40 @@ def epilogos_bedline_to_vector(bedline):
...
+        A line form a bedfile broken up into its contituent parts
+        (e.g. ["chr1", "1000", "2000", "state"]))
+
+
+    states_dic:
+        A dictionary containing the states in the file with a corresponding value
```
Comment (pkerpedjiev) [inline clodius/cli/convert.py]: An example like for the previous parameter would be nice.

File: clodius/cli/convert.py [hunk 3]
```diff
@@ -39,9 +39,40 @@ def epilogos_bedline_to_vector(bedline):
...
+
+    Returns
+    -------
+
+    An array containing values associated the state
```
Comment (pkerpedjiev) [inline clodius/cli/convert.py]: An example like the for the first parameter would be helpful here.

File: clodius/cli/convert.py [hunk 4]
```diff
@@ -39,9 +39,40 @@ def epilogos_bedline_to_vector(bedline):
...
+
+    '''
+    parts = bedline.decode('utf8').strip().split('\t')
```
Comment (pkerpedjiev) [inline clodius/cli/convert.py]: if `bedline` is an array as it says in the parameter definition above, the `.decode` function shouldn't work. I'm guessing that `bedline` is actually a string?

Comment (Lmercadom) [reply clodius/cli/convert.py]: Yeah,  bedline is just the line from the bedfile.
I meant that the function returns at the end an array of values associated with that line like chrom, start, end, states_vector. I will give a more specific example.

**Comment by Lmercadom** [comment]
I just submitted a PR for higlass adding the new data format in the documentation https://github.com/higlass/higlass/pull/551

**Pull request merged**